src/test/java/com/myhome/controllers/unit/AmenityControllerTest.java

```diff
@@ -154,9 +162,46 @@ void shouldNotAddAmenityWhenCommunityNotExists() {
     assertEquals(HttpStatus.NOT_FOUND, response.getStatusCode());
   }
 
+  @Test
+  void shouldReturnListOfBookingsOfGivenAmenityId() {
+    AmenityBookingItem amenityBookingItem = new AmenityBookingItem();
+    amenityBookingItem.setAmenity(getTestAmenity());
+    LocalDateTime currentDate = LocalDateTime.now();
+    GetAmenityBookingsResponse getAmenityBookingsResponse = new GetAmenityBookingsResponse();
+    getAmenityBookingsResponse.setAmenityId(TEST_AMENITY_ID);
+
+    List<AmenityBookingItem> amenityBookingItems = Collections.singletonList(amenityBookingItem);
+    given(amenitySDJpaService.listAllAmenityBookings(TEST_AMENITY_ID, currentDate, currentDate.plusDays(5), null))
+            .willReturn(amenityBookingItems);
+
+    given(amenityBookingItemApiMapper.amenityBookingToAmenityBookingsResponse(amenityBookingItems))
+            .willReturn(Collections.singletonList(getAmenityBookingsResponse));
+
+    ResponseEntity<List<GetAmenityBookingsResponse>> amenitiesBookings =
+            amenityController.getAmenitiesBookings(TEST_AMENITY_ID, currentDate, currentDate.plusDays(5), null);
+
+    assertEquals(HttpStatus.OK, amenitiesBookings.getStatusCode());
+    assertNotNull(amenitiesBookings.getBody());
+    assertEquals(1, amenitiesBookings.getBody().size());
+  }
+
+  @Test
+  void shouldReturn404WhenNoBookingsFound() {
+    LocalDateTime currentDate = LocalDateTime.now();
```

Comment:
Can you add test cases when startDate or endDate is null? What will happen then?